libraries/RC_Channel/RC_Channels_VarInfo.h

```diff
@@ -87,7 +90,7 @@ const AP_Param::GroupInfo RC_Channels::var_info[] = {
     // @Description: RC input options
     // @User: Advanced
     // @Bitmask: 0:Ignore RC Receiver, 1:Ignore MAVLink Overrides, 2:Ignore Receiver Failsafe, 3:FPort Pad, 4:Log RC input bytes, 5:Arming check throttle for 0 input, 6:Skip the arming check for neutral Roll/Pitch/Yay sticks, 7:Allow Switch reverse
-    AP_GROUPINFO("_OPTIONS", 33, RC_CHANNELS_SUBCLASS, _options, 0),
+    AP_GROUPINFO("_OPTIONS", 33, RC_CHANNELS_SUBCLASS, _options, (const float)RC_Channels::Option::ARMING_CHECK_THROTTLE),
```

Comment:
The type of _options is AP_Int32. Why is it a float type?

Reply:
Does seem strange to designate a bitmask slot with a float, but try compiling without that:
```
In file included from ../../ArduPlane/RC_Channel.cpp:11:0:
../../libraries/RC_Channel/RC_Channels_VarInfo.h:96:1: error: cannot convert ‘RC_Channels::Option’ to ‘const float’ in initialization
 };
 ^
```

Reply:
Its because its an enum class, you should cast to uint32_t

Reply:
Then why does the compiler say it wants a const float?

Reply:
Thanks, @muramura and @andyp1per 
casting to uint32_t works

Reply:
But doesn't this just obscure the fact that the type of def_value is "const float" ???

Reply:
It's not - it's AP_Int32. I suspect the warning is just an artifact of the fun and games used to implement this without much typing.

Reply:
Doesn't that compiler error I posted above says that the value is stored as a float in struct GroupInfo?
https://github.com/kd0aij/ardupilot/blob/44ed540fd24f2749470b7d3b15cd6eccc6a541de/libraries/AP_Param/AP_Param.h#L159

Reply:
I think this is relevant to the float type issue:
https://github.com/ardupilot/ardupilot/blob/8bfda973dbeba8c3e2480301ab694220011c5957/libraries/AP_Param/AP_Param.cpp#L1943